README.md

```diff
@@ -74,3 +74,47 @@ fn main() {
     }
 }
 ```
+
+Testing
+-------
+
+The driver test suite is largely composed of a mixture of integration tests and behavioral unit-tests, relying on the official [MongoDB specifications repo](https://github.com/mongodb/specifications). The easiest way to thoroughly test the driver is to set your fork up with TravisCI.
+
+To test the driver locally, you'll need to setup integration tests and specification tests.
+
+> NOTE: Each integration test uses a unique database/collection to allow tests to be parallelized, and will drop their dependencies before running. However, effects are _not_ cleaned up afterwards.
+
+#### Setting up integration tests
+
+All integration tests run on the default MongoDB port, 27017. Before running the tests, ensure that you setup a test database on that port.
+
+Download a version of MongoDB:
+```
+wget http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-3.2.11.tgz
+tar xvf mongodb-linux-x86_64-3.2.11.tgz
+mv mongodb-linux-x86_64-3.2.11 3.2.11
+```
+
+Run a MongoDB server on 27017:
+```
+mkdir -p ./data/test_db_3_2
+3.2.11/bin/mongod --fork --nopreallocj --dbpath ./data/test_db_3_2 --syslog --port 27017
```

Comment:
I'd change this to just `mongod --dbpath ./data/test_db_3_2`; I'm guessing most users running the tests won't want to continue running the server in the background after the tests are finished, `--syslog` is only necessary if you're using `--fork`, and `--nopreallocj` doesn't do anything on WiredTiger, which is the default in 3.2. (On 3.0 and earlier, it's only really worth if if the startup time of the database is too long for you for some reason, but given that the tests will likely end up needing to allocate the data files anyways, it's probably better not to use it there either. I took out the option in the Travis config file in #175 for similar reasons, as explained [here](https://github.com/mongodb-labs/mongo-rust-driver-prototype/pull/175/files#r94277292)).

Also, similar to above, it might be worth mentioning that you can just run `mongod ...` instead of `3.2.11/bin/mongod` if you already have it installed.